contracts/modules/Payments.sol

```diff
@@ -101,7 +101,9 @@ abstract contract Payments is RouterImmutables {
         }
         if (amount > 0) {
             WETH9.deposit{value: amount}();
-            WETH9.transfer(recipient, amount);
+            if (recipient != address(this)) {
```

Comment:
interesting, i'm struggling to think of a path where this would save gas - is there a case where we can't wrap ETH + transfer WETH from within a v3 callback?

Reply:
we chatted on slack - but basically this is because we separate out wrap/unwrap from trading so that the command can be used for NFT trades and other situations too. Also useful for when we have split trades on uniswap to only have to do a single wrap